src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -145,12 +152,26 @@ public T getValue() {
      *   List expected = asList(new Person("John"), new Person("Jane"));
      *   assertEquals(expected, peopleCaptor.getAllValues());
      * </code></pre>
+     *
+     * Example using custom matcher:
+     * <pre class="code"><code class="java">
+     *   mock.add(new ArrayList<>());
+     *   mock.add(new HashSet<>());
+     *   mock.add(LinkedList<>()):
+     *
+     *   ArgumentCaptor&lt;List&gt; listCaptor = ArgumentCaptor.forClass(List.class);
+     *
+     *   verify(mock, times(2)).add(peopleCaptor.captureIf(new InstanceOf(List.class));
+     *
+     *   List expected = asList(ArrayList<>(), LinkedList<>());
+     *   assertThat(expected.containsAll(peopleCaptor.getAllValues()), is(true));
+     * </pre>
      * See more examples in javadoc for {@link ArgumentCaptor} class.
      * 
      * @return captured argument value
      */
     public List<T> getAllValues() {
-        return this.capturingMatcher.getAllValues();
+        return capturingMatcher.getAllValues();
```

Comment:
Yes thats right for `getLastValue()`, `CapturingMatcher.getValues()` returns the plain arguments-List.